src/interpolations.jl

```diff
@@ -1535,6 +1536,95 @@ reference_coordinates(ip::VectorizedInterpolation) = reference_coordinates(ip.ip
+    end
+end
+
+n_components(::MatrixizedInterpolation{vdim1, vdim2}) where {vdim1, vdim2} = vdim1*vdim2
+adjust_dofs_during_distribution(ipm::MatrixizedInterpolation) = adjust_dofs_during_distribution(ipm.ip)
+
+# Vectorize to reference dimension by default
+function MatrixizedInterpolation(ip::ScalarInterpolation{shape}) where {refdim, shape <: AbstractRefShape{refdim}}
+    return MatrixizedInterpolation{refdim,refdim}(ip)
+end
+
+Base.:(^)(ipv::VectorizedInterpolation{vdim1}, vdim2::Int) where {vdim1} = MatrixizedInterpolation{vdim1, vdim2}(ipv)
+function Base.literal_pow(::typeof(^), ipv::VectorizedInterpolation{vdim1}, ::Val{vdim2}) where {vdim1,vdim2}
+    return MatrixizedInterpolation{vdim1, vdim2}(ipv.ip)
+end
+
+function Base.show(io::IO, mime::MIME"text/plain", ipm::MatrixizedInterpolation{vdim1, vdim2}) where {vdim1, vdim2}
+    show(io, mime, ipm.ip)
+    print(io, "^", vdim1 , "×", vdim2)
+end
+
+# Helper to get number of copies for DoF distribution
+get_n_copies(::MatrixizedInterpolation{vdim1, vdim2}) where {vdim1, vdim2} = vdim1*vdim2
+
+function getnbasefunctions(ipm::MatrixizedInterpolation{vdim1, vdim2}) where {vdim1, vdim2}
+    return vdim1 * vdim2 * getnbasefunctions(ipm.ip)
+end
+function shape_value(ipm::MatrixizedInterpolation{vdim, vdim, shape}, ξ::Tensors.Vec{refdim, T}, I::Int) where {vdim, refdim, shape <: AbstractRefShape{refdim}, T}
+    # First flatten to vector
+    i0, c0 = divrem(I - 1, vdim^2)
+    i = i0 + 1
+    v = shape_value(ipm.ip, ξ, i)
+
+    # Then compute matrix index
+    ci0, cj0 = divrem(c0, vdim)
+    ci = ci0 + 1
+    cj = cj0 + 1
+    return Tensor{2, vdim, T}((k, l) -> k == cj && l == ci ? v : zero(v))
+end
+
+function shape_value(ipm::MatrixizedInterpolation{vdim1, vdim2, shape}, ξ::Tensors.Vec{refdim, T}, I::Int) where {vdim1, vdim2, refdim, shape <: AbstractRefShape{refdim}, T}
+    # First flatten to vector
+    i0, c0 = divrem(I - 1, vdim1*vdim2)
+    i = i0 + 1
+    v = shape_value(ipm.ip, ξ, i)
+
+    # Then compute matrix index
+    ci0, cj0 = divrem(c0, vdim1)
+    ci = ci0 + 1
+    cj = cj0 + 1
+
+    return SMatrix{vdim1, vdim2, T}(ntuple(i_ -> i_ == (cj0*vdim2 + ci) ? v : zero(v), vdim1*vdim2))
+end
+
+# vdim1 == vdim2 == refdim
+function shape_gradient_and_value(ipm::MatrixizedInterpolation{dim, dim, shape}, ξ::Vec{dim}, I::Int) where {dim, shape <: AbstractRefShape{dim}}
+    return invoke(shape_gradient_and_value, Tuple{Interpolation, Vec, Int}, ipm, ξ, I)
+end
+
+# vdim1 != vdim2 != refdim
+function shape_gradient_and_value(ipm::MatrixizedInterpolation{vdim1, vdim2, shape}, ξ::V, I::Int) where {vdim1, vdim2, refdim, shape <: AbstractRefShape{refdim}, T, V <: Vec{refdim, T}}
+    tosmat(v::Tensor{2}) = SMatrix{vdim1,vdim2}((v...,))
+    tosmat(v::SMatrix) = v
+    tosvec(v::Vec) = SVector((v...,))
+    tovec(sv::SVector) = Vec((sv...))
+    val = shape_value(ipm, ξ, I)
+    grad = ForwardDiff.jacobian(sv -> tosmat(shape_value(ipm, tovec(sv), I)), tosvec(ξ))
```

Comment:
I don't think this will do the right thing unfortunately.
But for these matrix interpolations, it probably is better to calculate the gradients wrt. the scalar base-interpolation, and then just insert the values in the right places (and zeros elsewhere).

Reply:
Do you have any suggestion how we can cover this case in testing (in addition to the vectorial case)?

Reply:
I think @lijas solution [here](https://github.com/Ferrite-FEM/Ferrite.jl/blob/cf97ee510936259a22ab0e9dee0e423f918112e6/test/test_interpolations.jl#L225-L234) was quite nice, just having the option of calling that function explicitly as setup [here](https://github.com/Ferrite-FEM/Ferrite.jl/blob/cf97ee510936259a22ab0e9dee0e423f918112e6/src/interpolations.jl#L1533-L1540). 

Maybe combined with calling something like 
```julia
for Is in 1:getnbasefunctions(ipm.ip)
    g, v = shape_gradient_and_value(ipm.ip, ξ, Is)
    Im = (Is - 1)*getnbasefunctions(ipm.ip)
    for i in 1:vdim1, j in 1:vdim2
        Im += 1
        G, V = shape_gradient_and_value(ipm.ip, ξ, Im)
        @test G[i,j,:] \approx g
        @test V[i,j] \approx v
    end
end
```